vertexai/src/VertexAI.cs

```diff
@@ -15,33 +15,96 @@
  */
 
 using System;
+using System.Collections.Concurrent;
 
 namespace Firebase.VertexAI {
 
+/// <summary>
+/// The entry point for all Vertex AI in Firebase functionality.
+/// </summary>
 public class VertexAI {
 
-  public static VertexAI DefaultInstance {
-    get {
-      throw new NotImplementedException();
-    }
+  private static readonly ConcurrentDictionary<string, VertexAI> _instances = new();
+
+  private FirebaseApp _firebaseApp;
+  private string _location;
+
+  private VertexAI(FirebaseApp firebaseApp, string location) {
+    _firebaseApp = firebaseApp;
+    _location = location;
   }
 
+  /// <summary>
+  /// Returns a VertexAI initialized with the default `FirebaseApp` and location.
+  /// </summary>
+  public static VertexAI DefaultInstance => GetInstance();
+
+  /// <summary>
+  /// Creates an instance of `VertexAI` configured with the default `FirebaseApp`, and the given location.
+  /// </summary>
+  /// <param name="location">The region identifier, defaulting to `us-central1`; see [Vertex AI
```

Comment:
I wonder if we should hard code the default value here, as who knows if it might change in the backend long after we've thought about this code.

Reply:
Unfortunately, the location is used to talk to the correct endpoint, so it has to be hardcoded somewhere, and this is how the other platforms do it. And yeah, there are already talks about it changing :/

Reply:
👍